### PR TITLE
[7172] Added max length for field lengths summary

### PR DESCRIPTION
**Pull request by defong**
### Context
Field lengths summary

### Changes proposed in this pull request
Added max length for field lengths summary

### Guidance to review
It makes senses from /api-docs/reference/Register_API_Reference_v0.1.xlsx

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/views/api_docs/reference/v0.1/_reference.md [hunk 1]
```diff
@@ -2368,7 +2370,7 @@ Deletes an existing degree for this trainee.
     <dt class="govuk-summary-list__key"><code>previous_last_name</code></dt>
     <dd class="govuk-summary-list__value">
       <p class="govuk-body">
-        string
+        string (limited to 60 characters)
```
[the remark this answers is not in the document]
Comment (defong) [reply app/views/api_docs/reference/v0.1/_reference.md:2373]: Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/4315/commits/f113945c610818a2604d3ef3d3cb11c1ee6abfec

File: app/views/api_docs/reference/v0.1/_reference.md [hunk 2]
```diff
@@ -2354,7 +2356,7 @@ Deletes an existing degree for this trainee.
     <dt class="govuk-summary-list__key"><code>last_name</code></dt>
     <dd class="govuk-summary-list__value">
       <p class="govuk-body">
-        string, required
+        string (limited to 60 characters), required
```
[the remark this answers is not in the document]
Comment (defong) [reply app/views/api_docs/reference/v0.1/_reference.md:2359]: Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/4315/commits/f113945c610818a2604d3ef3d3cb11c1ee6abfec

File: app/views/api_docs/reference/v0.1/_reference.md [hunk 3]
```diff
@@ -2340,7 +2342,7 @@ Deletes an existing degree for this trainee.
     <dt class="govuk-summary-list__key"><code>first_names</code></dt>
     <dd class="govuk-summary-list__value">
       <p class="govuk-body">
-        string, required
+        string (limited to 60 characters), required
```
[the remark this answers is not in the document]
Comment (defong) [reply app/views/api_docs/reference/v0.1/_reference.md:2345]: Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/4315/commits/f113945c610818a2604d3ef3d3cb11c1ee6abfec

File: app/views/api_docs/reference/v0.1/_reference.md [hunk 4]
```diff
@@ -2841,7 +2843,7 @@ Deletes an existing degree for this trainee.
     <dt class="govuk-summary-list__key"><code>uk_degree</code></dt>
     <dd class="govuk-summary-list__value">
       <p class="govuk-body">
-        string, required if degree is from the UK
+        string (limited to 6 characters), required if degree is from the UK
```
[the remark this answers is not in the document]
Comment (defong) [reply app/views/api_docs/reference/v0.1/_reference.md]: Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/4315/commits/a6c7a0965d6436b59d001c67accbc2542887c6e8

File: app/views/api_docs/reference/v0.1/_reference.md [hunk 5]
```diff
@@ -2312,7 +2314,7 @@ Deletes an existing degree for this trainee.
     <dt class="govuk-summary-list__key"><code>application_id</code></dt>
     <dd class="govuk-summary-list__value">
       <p class="govuk-body">
-        string
+        string (limited to 10 characters)
```
[the remark this answers is not in the document]
Comment (tomtrentham) [reply app/views/api_docs/reference/v0.1/_reference.md:2317]: I've updated the implementation ticket for this field https://trello.com/c/YE5kYr5I/60-add-support-for-applicationid

**Comment by tomtrentham** [comment]
> I think the length limits are less necessary on enum or inclusion list type values because if it's in the list it's guaranteed to be the right length. We can leave the max lengths in if it makes consumers of the API happy but it's one more thing to maintain for us.

@stevehook Yes, it's a good point, particularly for HESA data where the API users should know the field sizes.

For Register data, I think it will be useful for API users to have a quick reference point for how large the field needs to be if they are adding it to their systems. One more thing for us to keep up to date though.

**Review by tomtrentham — APPROVED** [review]
LGTM

**Pull request merged**